### ISSUE TITLE
Is there pytorch version?

### ISSUE THREAD

**Issue by beiluo-horizon** [Qcactus/CPR#1]
Hi, thanks for your excellent work, Is there pytorch version?

**Comment by beiluo-horizon** [comment]
In additon, I set k_interact to 2, sample_rate and sample_ratio to 3, and batch_size to 2048. According to the description in the article, each batch should eventually include 4096 users, but the code actually contains 12288 users. Why does this happen?

**Comment by Qcactus** [comment]
Sorry. We don't have a pytorch version.

About the number of users, you can have a look at **Algorithm 1** in our paper. We sample `batch_size*sample_rate` samples (line 6), and then keep the hardest `batch_size` samples for training (line 9) – this is done [here](https://github.com/Qcactus/CPR/blob/7d208de954a3bf47820113566b6b4300cf090496/recq/utils/loss.py#L25) in codes.

**Comment by beiluo-horizon** [comment]
thanks for your prompt reply! This is exactly why I am confused. in cpy.py file, there is code:
for users, items in self.sampler.sample():
I am Looking at the shape of users after this line, I observed that its dimension is 12288, which is consistent with batch* sampling rate*k, the result is consistent with line 6 in Algorithm 1. But after 7-10 lines of Algorithm 1, shouldn't its dimension be consistent with batch*k?

**Comment by Qcactus** [comment]
[Here](https://github.com/Qcactus/CPR/blob/7d208de954a3bf47820113566b6b4300cf090496/recq/recommenders/cpr.py#L74) `self.mf_loss` is calculated [only by `batch_size` samples](https://github.com/Qcactus/CPR/blob/7d208de954a3bf47820113566b6b4300cf090496/recq/utils/loss.py#L25-L26).
